cisco_ise/config.schema.yaml

```diff
@@ -0,0 +1,30 @@
+---
+connections:
+  type: object
+  required: true
+  patternProperties:
+    "^\\w+":
+      "$ref": "#/properties/connection_name"
+  additionalProperties: false
```

Comment:
I think it would be better to put these properties under additionalProperties.
https://github.com/StackStorm/st2/pull/5225 fixed the ability to retrieve encrypted secrets from the datastore under `additionalProperties`. I don't think anything checks for `patternProperties`.

Modifying your example, I don't think this will work:

```yaml
connections:
  ise_server:
    api_user: username
    api_pass: "{{ st2kv.system.ise_server_api_pass }}"
    hostname: ise-pan.example.com
```

Also, what happens if someone does this? Technically I think passes the schema (possibly needs one entry in connections):
```yaml
connections: {}
connection_name:
  api_user: username
  api_pass: password
  hostname: ise-pan.example.com
```

Reply:
Good feedback, I will give this a review and make some adjustments.

Reply:
I'm having trouble with how this should be done to make it better, would you be willing to offer a suggestion?

Reply:
Here's how I did it for the vault pack (not upstreamed yet): https://github.com/copartit/stackstorm-vault/blob/dev/config.schema.yaml#L20

Reply:
I think for now I would prefer to leave it as is. I know it at least works as intended and I'm not sure I'll have time to modify and test it properly. Hopefully that is ok.